src/core/src/node_input.cpp

```diff
@@ -9,6 +9,12 @@
 namespace ov {
 Input<Node>::Input(Node* node, size_t index) : m_node(node), m_index(index) {}
 
+Input<Node>::Input(const std::shared_ptr<Node>& node, size_t index) : m_node(node.get()), m_index(index) {}
```

Comment:
Input<Node> class doesn't extend life for shared_ptr<Node>,  so  only Output<Node> has c-tor which takes shared_ptr as input.